### PR TITLE
Fix agent memory usage check

### PR DESCRIPTION
**Pull request by nagworld9**
## Description
Added fix for addressing agent reverted to old version due to multiple restarts too frequently because of agent exits with memory limit reach.

We revert to old version if the agent restarts 3 times in CHILD_LAUNCH_INTERVAL. That's why we delay first attempt memory usage check, so that it won't meet this condition on restart.

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/ga/update.py [hunk 1]
```diff
@@ -1016,8 +1017,11 @@ def _check_agent_memory_usage(self):
         """
         try:
             if conf.get_enable_agent_memory_usage_check() and self._extensions_summary.converged:
-                if self._last_check_memory_usage == datetime.min or datetime.utcnow() >= (self._last_check_memory_usage + UpdateHandler.CHECK_MEMORY_USAGE_PERIOD):
-                    self._last_check_memory_usage = datetime.utcnow()
+                # we delay first attempt memory usage check, so that current agent won't get blacklisted due to multiple restarts(because of memory limit reach) too frequently
+                if (self._initial_attempt_check_memory_usage and time.time() - self._last_check_memory_usage_time > CHILD_LAUNCH_INTERVAL) or \
```
Comment (narrieta) [inline azurelinuxagent/ga/update.py:1021]: why the change from utcnow() to time()? - the former has the advantage that it formats nicely when, for example, running the code under the debugger

**Review by narrieta — APPROVED** [review]

File: azurelinuxagent/ga/update.py (hunk 1 above)
Comment (nagworld9) [reply azurelinuxagent/ga/update.py:1021]: Nothing specific, I just came across the idea. Let me update if it has advantage

Comment (narrieta) [reply azurelinuxagent/ga/update.py:1021]: ok, it is not a big issue.

**Review by maddieford — APPROVED** [review]
